core/amber/src/main/scala/edu/uci/ics/texera/web/service/WorkflowService.scala

```diff
@@ -120,7 +119,7 @@ class WorkflowService(wid: String, cleanUpTimeout: Int) extends LazyLogging {
     )
     jobService = Some(state)
     jobStateSubject.onNext(state)
-    state.startWorkflow()
+    state.startWorkflow(wid.split('-')(1).toInt)
```

Comment:
This is too ad-hoc. I think we can utilize the existing `WorkflowContext` here. Try to replace the wid with `WorkflowContext` then pass it to `WorkflowJobService`

Reply:
Thanks for the suggestion. I have a question. 
1. an instance of `WorkflowContext` is only created in `WorkflowJobService`. My understanding is that you want this instance to be created earlier? in `WorkflowService`?
2. `WorkflowContext` doesn't have the `wid` information, I still have to extract it from the string as above. Are you suggesting I still extract it the same way and add the information to `WorkflowContext`?